src/sockperf.cpp

```diff
@@ -1713,12 +1713,33 @@ static int resolve_sockaddr(const char *host, const char *port, int sock_type,
     // any protocol
     hints.ai_protocol = 0;
     hints.ai_socktype = sock_type;
-
+    int res;
     struct addrinfo *result;
-    int res = getaddrinfo(host, port, &hints, &result);
+    std::string path = host;
+
+    if (path.size() > 0 && path[0] == '/') {
+        log_dbg("provided path for Unix Domain Socket is %s\n", host);
+        if (path.length() > MAX_UDS_NAME) {
+            log_err("length of name is larger than %d bytes", MAX_UDS_NAME);
+            res = SOCKPERF_ERR_SOCKET;
+            return res;
+        }
+        addr_len = path.length() + 2;
+        strncpy(((sockaddr_store_t*)addr)->addr_un.sun_path, path.c_str(), addr_len);
+        ((sockaddr_store_t*)addr)->addr_un.sun_family = AF_UNIX;
+        ((sockaddr_store_t*)addr)->ss_family = AF_UNIX;
+
+        return 0;
+    }
+    res = getaddrinfo(host, port, &hints, &result);
     if (res == 0) {
         get_socket_address(result, addr, addr_len);
+    } else if (res == EAI_NONAME && host != NULL) { // unix case but didn't start with '/'
+        log_err("unix domain socket path have to start with '/'");
+        res = SOCKPERF_ERR_SOCKET;
+        return res;
```

Comment:
This also can happen when DNS server is not available, so it will be strange to get message about UNIX socket in this case. I think that it's better to just return an error without any hints.